packages/blade/src/components/Popover/_decisions/decisions.md

```diff
@@ -0,0 +1,275 @@
+| titleLeading   | `React.ReactNode`                                                                                                      | `undefined` | Leading element to be placed before header title, recommended components are Icon or Asset. The icon's size will be set to large      |          |
+| footer   | `React.ReactNode`                                                                                                      | `undefined` | Slottable content of the footer                                                                                                       |          |
+| content         | `React.ReactNode`                                                                                                               | `undefined` | Content of the popover (can be any interactive or non-interactive elements)                                                                                                                |  ✅        |
+| children        | `React.ReactNode`                                                                                                      | `undefined` | Trigger component for popover, Accepts any interactive element or icons                                                               | ✅        |
+| placement       | `top, top-start, top-end, left, left-start, left-end, right, right-start, right-end, bottom, bottom-start, bottom-end` | `top`       | Placement of popver, the popver avoid collision with the edge of the screen and flip to the oppisite side even when placement is set. |          |
+| isOpen          | `boolean`                                                                                                              | `undefined` | Controlled state of the popover                                                                                                       |          |
+| defaultIsOpen   | `boolean`                                                                                                              | `undefined` | Uncontrolled state of the popover                                                                                                     |          |
+| onOpenChange    | `({ isOpen }) => void`                                                                                                 | `undefined` | Called when popover isOpen state is changed, this can be used to detect when popover opens or closed                                  |          |
+| onDismissButtonClick    | `() => void`                                                                                                 | `undefined` | Called when dismiss button is clicked, useful for sending analytic events                                  |          |
```

Comment:
Separate event handler for dismiss seems bit redundant because we already have an event handler that runs for both show / hide 🤔

Reply:
Yeah kinda agree 🤔  could cause confusion actually, we added onDismissButtonClick for analytics purpose. https://github.com/razorpay/blade/pull/1584#discussion_r1315529213

@kamleshchandnani maybe we can re-evaluate this, the question is what is the intent that analytics team will be interested in? It will most probably be "closing of popup" doesn't matter how. 

Plus it's more convenient to click the popup by clicking outside than clicking the dismiss button, thus if someone adds analytic events in onDismissButtonClick, the analytics team might not get all the "closed popup" events.

Reply:
Did we decide anything on this?

Reply:
While there are trade-offs the thing is product when it comes to analytics they track everything. But we can comeback on this and might add this as an enhancement. Think of it this way if product wants to track ho many people dismissed the popover via moving the cursor outside vs clicking on the dismiss button and that can change a lot of things for product. but lets park it for now